src/main/java/org/disq_bio/disq/impl/formats/bgzf/BgzfBlockSource.java

```diff
@@ -67,7 +67,7 @@ private BgzfBlockGuesser getBgzfSplitGuesser(Configuration conf, String path) th
 
       @Override
       protected BgzfBlock advance() {
-        if (start > splitEnd) {
+        if (start >= splitEnd) { // splitEnd is exclusive, so if we reach it then stop
```

Comment:
If i'm not mistaken, this was the actual fix correct? Everything else in this branch seems to revolve around extra logging. My next question, what are the performance implications of log4j loggin being done per-executor? Does it have to call back to master in any capacity or will it kill debug messages before they get off the groud?